src/test/java/org/junitpioneer/jupiter/resource/ResourcesParallelismTests.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+import static org.junit.jupiter.api.parallel.ExecutionMode.SAME_THREAD;
+import static org.junitpioneer.jupiter.resource.Scope.GLOBAL;
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import org.assertj.core.api.Condition;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+class ResourcesParallelismTests {
+
+	@DisplayName("when a number of shared resources are used in the same test suite")
+	@Nested
+	class WhenANumberOfSharedResourcesAreUsedInSameTestSuiteTests {
+
+		@DisplayName("then the tests do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestsDoNotRunInParallel() {
+			ExecutionResults executionResults = assertTimeoutPreemptively(Duration.ofSeconds(15),
+				() -> PioneerTestKit.executeTestClass(ThrowIfTestsRunInParallelTestCases.class),
+				"The tests in ThrowIfTestsRunInParallelTestCases became deadlocked!");
+			assertThat(executionResults).hasNumberOfSucceededTests(3);
+		}
+
+		@DisplayName("then the test factories do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestFactoriesDoNotRunInParallel() {
```

Comment:
I don't think we need to test this as an extension doesn't care whether it's operating on a test or test factory, right?

Reply:
My thinking was that test factories could have a `@New` or `@Shared`-annotated parameters, just like normal tests, so I wanted to check that test factories annotated with `@Shared` executed sequentially too.

Is my thinking incorrect somewhere?

Reply:
I got that but I thought it doesn't matter to our extension because we don't interact with that. But we very much do when we implement `InvocationInterceptor`, particularly `interceptTestMethod`, `interceptTestFactoryMethod`, `interceptDynamicTest`, and `interceptTestTemplateMethod`.

Reply:
Great, I'm glad I didn't miss anything here. Thanks, Nicolai. :)